### PR TITLE
Add CI for Windows with Appveyor

### PR DESCRIPTION
**Pull request by xiamx**
See #6141 and https://github.com/elixir-lang/elixir/pull/6141#issuecomment-303732946

Here's what it looks like:
https://ci.appveyor.com/project/xiamx/elixir/history

Status badge:
https://www.appveyor.com/docs/status-badges/

### CODE REVIEW

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 1.4.4-{build}
```
Comment (ericmj) [inline appveyor.yml]: Is this needed? Do we need to update this for every release or can we automate it?

**Comment by xiamx** [comment]
We could probably just have
`version: {build}‘


MengXuan Xia

On May 24, 2017, at 10:49 AM, Eric Meadows-Jönsson <notifications@github.com<mailto:notifications@github.com>> wrote:

version: 1.4.4-{build}

**Comment by robi-wan** [comment]
I'll prepare a PR. But it will take some time until I am back at a computer.

**Comment by robi-wan** [comment]
@xiamx `--ignore-errors` is maybe not what we want - this really ignores errors which means the exit/return code is 0. `--keep-going` seems to be a better option. But there seems to be an issue with AppVeyor: using this options does not end the build - it hangs. I have to manually cancel the build:
https://ci.appveyor.com/project/robi-wan/elixir/build/1-windows-ci+2

But at least setting the PATH works: https://github.com/robi-wan/elixir/commit/0a195bf3b0a492e9ec27642dae1f17153f8bf531#diff-11c909939117928998b102a1fff7d363 ;-)

**Comment by xiamx** [comment]
@robi-wan  

>But there seems to be an issue with AppVeyor: using this options does not end the build - it hangs. I have to manually cancel the build:

Yes, I discovered this issue is even present without `--keep-going` or `--ignore-errors`. `make test` is leaking `erl.exe` even though it terminates with exit code 1. I'll make submit an issue to track this..

**Comment by xiamx** [comment]
I also had some work in progress in my fork https://github.com/xiamx/elixir/tree/windows_compatibility

**Pull request merged**